### PR TITLE
refactor: simplify Database retry backoff logic (#3368)

### PR DESCRIPTION
**Pull request by tustvold**
Whilst working on #3368 I realised this retry loop could be simplified

### CODE REVIEW

File: server/src/database/init.rs [hunk 1]
```diff
@@ -616,62 +607,15 @@ pub(crate) async fn initialize_database(shared: &DatabaseShared) {
...
 
-    // error throttling
-    // - checks if the current error was already throttled
-    // - keeps a backoff duration that will change over the course of multiple errors
-    let mut throttled_error = false;
+    // A backoff duration for retrying errors that will change over the course of multiple errors
     let mut backoff = INIT_BACKOFF;
 
     while !shared.shutdown.is_cancelled() {
-        // Acquire locks and determine if work to be done
```
Comment (tustvold) [inline server/src/database/init.rs:626]: I found this block quite hard to follow as it is potentially waiting for the error retry of the previous iteration, by moving it to the end of the while loop I think it is easier to follow

File: server/src/database/init.rs [hunk 2]
```diff
@@ -744,6 +683,37 @@ pub(crate) async fn initialize_database(shared: &DatabaseShared) {
             *state.unfreeze(handle) = next_state;
             shared.state_notify.notify_waiters();
         }
+
```
Comment (crepererum) [inline server/src/database/init.rs:686]: Just checking: in this case did we already drop all the locks from the DB (including read locks) so gRPC requests are free to modify the state?

Comment (crepererum) [reply server/src/database/init.rs:686]: If the answer is "yes, all locks gone", feel free to merge it
